src/ui-scroll.js

```diff
@@ -397,7 +400,7 @@ angular.module('ui.scroll', [])
           // We need the item bindings to be processed before we can do adjustments
           !$scope.$$phase && !$rootScope.$$phase && $scope.$digest();
 
-          if (!rowHeight) {
+          if (allowVisibilityWatch) {
```

Comment:
Showing new rows after binding relates to hiding new rows before binding. So it does not relate to visibility-watcher, it may relate to rowHeight (see previous comment).